cli/src/commands/compatibility.rs

```diff
@@ -108,7 +110,8 @@ async fn fetch_runtime_metadata(url: &Uri) -> color_eyre::Result<RuntimeMetadata
     }
 
     match metadata.1 {
-        RuntimeMetadata::V14(v14) => Ok(v14),
+        RuntimeMetadata::V14(v14) => Ok(metadata_v14_to_latest(v14)),
```

Comment:
I feel like the CLI tool should just give back whatever metadata version is available and not try to transform it (that way people can just use it to download whatever the latest metadata version is from a node rather than specifically returning something that subxt wants. Then, the subxt macro/codegen can do this update function instead to be compatible with as many versions as possible (whether downloaded via our cli tool or other)

Reply:
Yep, for internal functionality this PR uses the V15:
- retain only specific pallets
- validation hashing 
- codegen

Otherwise, we would need to handle both v14 and v15 internally, which would lead to extra code we'd have to maintain.

Transitioning from V14 to V15 shouldn't cause any breaking changes in the output, expect the output of `cli metadata --retain-pallets`.
And when users point to a V14 node and use plain `cli metadata` the outputted metadata would not be converted to V15.

Reply:
Ah yup; I missed that we preserve V14 if you don't use `--pallets` so that looks good to me as-is :)

let's avoid implementing a way to downgrade to V14 again at the mo